### ISSUE TITLE
[NodeCore 0.2.4] Crash after loading saved blockchain from disk

### ISSUE THREAD

**Issue by a123b** [VeriBlock/nodecore-releases#66]
I stopped nodecore "gracefully" (with only SIGINT) at height 39745. One of the last messages before closing itself was something along the lines of "Blockchain successfully saved."

When I started nodecore again, it started to load the blockchain up until block 39632 before it crashed with the output below. Note that it was over 100 blocks ahead of this block when it was saved.
```
18:57:32.503 [main] INFO  nodecore.core.a.j - 37641 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37640 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37639 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37638 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37637 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37636 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37635 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37634 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37633 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37632 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.j - 37631 is being archived!
18:57:32.503 [main] INFO  nodecore.core.a.m - Chain walk back and replay succeeded
18:57:32.504 [main] INFO  nodecore.core.a.m - About to save block...
18:57:32.504 [main] INFO  nodecore.core.a.m - Saved block...
18:57:32.504 [main] INFO  nodecore.core.a.m - Setting the local blockchain height...
18:57:32.505 [main] INFO  nodecore.core.a.m - Before adding block 39631(0000000004ED8A5B463F371B03BE37691C6132FBCD4224F5), latest Bitcoin block known: 000000000001B31C7C315CF8EACB2C79EE4CC2C1FD578F9C2358BDBB7822AF92
18:57:32.505 [main] INFO  nodecore.core.a.m - About to process block cache...
18:57:32.505 [main] INFO  nodecore.core.a.m - Done processing block cache...
18:57:32.505 [main] INFO  nodecore.core.a.m - About to add block directly...
18:57:32.505 [main] INFO  nodecore.core.a.m - Attempting to add block 0000000004ED8A5B463F371B03BE37691C6132FBCD4224F5 as an alternate chain
18:57:32.525 [main] INFO  nodecore.core.a.m - Added block directly... state: SUCCESSFULLY_ADDED_ALTERNATE_CHAIN_NOW_BEST
18:57:32.525 [main] INFO  nodecore.core.a.m - Added a new block (0000000004ED8A5B463F371B03BE37691C6132FBCD4224F5 @ 39631), new best block!
18:57:32.544 [main] INFO  nodecore.core.a.m - Found a new best blockchain, ending at block 39631 (0000000004ED8A5B463F371B03BE37691C6132FBCD4224F5)!
18:57:32.544 [main] INFO  nodecore.core.a.m - Found last common block between the two chains: 39629 -> 00000000068F3A5F2E7D1E6BADC4FFDB20BD082AE5B3C359
18:57:32.544 [main] INFO  nodecore.core.a.m - Unapplying block: 39631 -> 0000000002E849D177E188F3D6ECD0E94F3E2C7FE689F742
18:57:32.545 [main] INFO  nodecore.core.a.m - Ledger hash before unapplying block 39631: EB9B71DD5F02486184E976BBEAA3401CD9E351CE0DBB7503
18:57:32.548 [main] ERROR nodecore.core.a.m - resolveToNewBestChain failed
java.lang.IndexOutOfBoundsException: Index: 39632, Size: 39632
        at java.util.ArrayList.rangeCheck(ArrayList.java:653)
        at java.util.ArrayList.get(ArrayList.java:429)
        at nodecore.core.a.m.a(DefaultBlockchain.java:1434)
        at nodecore.core.a.m.c(DefaultBlockchain.java:1987)
        at nodecore.core.a.m.a(DefaultBlockchain.java:1878)
        at nodecore.core.a.m.h(DefaultBlockchain.java:203)
        at nodecore.Program.a(Program.java:84)
        at nodecore.Program.main(Program.java:156)
18:57:32.548 [main] ERROR nodecore.Program - Unable to load blockchain.
java.lang.IndexOutOfBoundsException: Index: 39632, Size: 39632
        at java.util.ArrayList.rangeCheck(ArrayList.java:653)
        at java.util.ArrayList.get(ArrayList.java:429)
        at nodecore.core.a.m.a(DefaultBlockchain.java:1434)
        at nodecore.core.a.m.c(DefaultBlockchain.java:1987)
        at nodecore.core.a.m.a(DefaultBlockchain.java:1878)
        at nodecore.core.a.m.h(DefaultBlockchain.java:203)
        at nodecore.Program.a(Program.java:84)
        at nodecore.Program.main(Program.java:156)
```

**Comment by a123b** [comment]
At least, after deleting the blockchain chunks from `396.bdt` onwards, it successfully loaded to block 39599 and continued to sync the rest

**Comment by Hou5e** [comment]
Same for me with nodecore-0.2.5:
[LoadingBlockchainError.txt](https://github.com/VeriBlock/nodecore-releases/files/2237458/LoadingBlockchainError.txt)

**Comment by a123b** [comment]
Can confirm the problem persists with nodecore-0.2.5, even if the blockchain was saved with 0.2.5.

**Comment by a123b** [comment]
Fixed in 0.2.6
